***3. Programming Practice***

**1. Write a function that takes a list of integers and calculates their sum.**

In [6]:
def summation(lst):
    sum = 0
    for num in lst:
        sum = sum+num
    return sum

In [7]:
x = input("Enter the list of integers: ")
l = list(map(int,x.split()))
summation(l)

Enter the list of integers: 7 8 9 6 84 39


153

**2. Write a function that takes a list of integers and calculates the sum of all the odd values. Return the sum and the number of values encountered.** 

In [8]:
def oddsum(numbers):
  summ = 0
  for num in numbers:
    if(num%2!=0):
      summ = summ + num
  return(summ)

In [9]:
y = input("Enter the list of integers: " )
numbs = list(map(int,y.split()))
oddsum(numbs)

Enter the list of integers: 1 9 8 7 10 5


22

**3. Determine the max and min of three numbers.**

In [22]:
def maxmin(x,y,z):
    if (x>y):
        max = x
        min = y
    else:
        max = y
        min = x
    if z>x:
        max = z
    elif z<y:
        min = z
    return max, min

In [23]:
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))
c = int(input("Enter third number: "))
maxmin(a,b,c)

Enter first number: 89
Enter second number: 86
Enter third number: 91


(91, 86)

**4. Determine if a given year is a leap year.**

In [14]:
def leapp(y):
    if (y%4!=0):
        return "Not a leap year"
    elif(y%100!=0):
        return "Leap year"
    elif(y%400==0):
        return "Leap year"
    else:
        return "Not a leap year"

In [18]:
def leapp(y):
    if (y%4!=0):
        return "Not a leap year"
    elif(y%100==0 and y%400==0 or y%100!=0):
        return "Leap year"
    else:
        return "Not a leap year"

In [29]:
def leapp(y):
    if(y%4==0 and y%100==0 and y%400==0 or y%4==0 and y%100!=0):
        return "Leap year"
    else:
        return "Not a leap year"

In [30]:
y = int(input("Enter the year: "))
leapp(y)

Enter the year: 1904


'Leap year'

**5. Implement factorial iteratively and recursively.**

In [30]:
def fact(n):
    f = 1
    for i in range(1,n+1):
        f = f*i
    return f

In [31]:
x = int(input("Enter the number for factorial: "))
fact(x)

Enter the number for factorial: 5


120

**6. Calculate the nth Fibonacci number iteratively and recursively.**

In [37]:
def fibo(n):
    fibo = [0,1]
    for i in range(2,n):
      a = fibo[i-2] + fibo[i-1]
      fibo.append(a)
    return fibo[-1]

In [38]:
x = int(input("Enter the nth Fibonacci number: "))
fibo(x)

Enter the nth Fibonacci number: 5


3

**7.Write a function that checks if a given word is a palindrome.**

In [5]:
def palindrome(word):
    i = 0
    j = len(word) - 1
    while(i<j):
        if (word[i]!= word[j]):
            return "Not a Palidnrome word"
        i = i + 1
        j = j - 1
    return "Palidnrome word"

In [6]:
s = str(input())
palindrome(s)

civic


'Palidnrome word'

**8. Generate a list of random integers. Take input 
 from the user (an integer value) and search the list for 
. Return the index of 
 in the list, if it exists, or None otherwise.**

In [47]:
import random
def rndm(rndnums, k, integer_number):
    integer_number = [random.randint(1, 100) for i in range(rndnums)]
    try:
        y = integer_number.index(k)
        print(y)
    except ValueError:
        print(None)

In [48]:
numss = int(input("Required numbers in the list: "))
m = int(input("Find the index of value: "))
integerss = [random.randint(1, 100) for i in range(numss)]
rndm(numss, m, integerss)

Required numbers in the list: 100
Find the index of value: 10
61


**9th. Calculate $e^x$, sin(x) and cos(x)recursively.**

In [15]:
import math
def trigo():
    x = int(input())
    degree = math.radians(x)
    c = 0
    while(c==0):
        ans  = round(math.exp(x) * 10**13,2)
        sinn = round(math.sin(degree),2)
        coss = round(math.cos(degree),2)
        print(ans)
        print(sinn)
        print(coss)
        c=c+1
    
trigo()

30
1.0686474581524462e+26
0.5
0.87


**10. Check if a Number is an Armstrong Number. An Armstrong number is a number that is the sum of its own digits each raised to the power of the number of digits. For example, 153 = 1^3 + 5^3 + 3^3.**

In [1]:
import math
def arm(x):
    y = len(x)
    summ = 0
    for i in x:
        cub = math.pow(int(i),y)
        summ = summ + cub
    if int(x) == summ:
        return 'Armstrong number'
    else:
        return 'Not an Armstrong number'

In [3]:
u = input("Enter a number to check: ")
arm(u)


'Not an Armstrong number'

In [1]:
def lnnum(n):
    c=0
    while(n>0):
        n = n//10
        c = c + 1
    return c

In [6]:
lnnum(15378)

5

In [18]:
# def arms(z,y):
#     arm = 0
#     while(z>0):
#         rem = z%10
#         arm = rem**y + arm
#         z = z//10
#     if(arm==z):
#         return True
#     else:
#         return False


In [19]:
# a = int(input("Enter a number"))
# x = lnnum(a)
# arms(a,x)

False

**11. Take a 3 digit number from the user, with different first and last digits. Reverse it and subtract the smaller from the larger. Call this d. Reverse d and add it to d. Print the result.**

In [1]:
def revv(num):
    if(len(num)==3 and num[0]!=num[2]):
        rev = num[-1::-1]
        print(rev)
        if(int(num)>int(rev)):
            d = int(num) - int(rev)
            d1 = str(d)
            revd = d1[-1::-1]
            revd1 = int(revd)
            ans = revd1 + d
            print(ans)
        else:
            d = int(rev) - int(num)
            d1 = str(d)
            revd = d1[-1::-1]
            revd1 = int(revd)
            ans = revd1 + d
            print(ans)
    else:
        print("Invalid Input")

In [2]:
z = input("Enter the three digit number: ")
revv(z)

Enter the three digit number: 153
351
1089


* Lenght of an integer

In [2]:
def lnnum(n):
    c=0
    while(n>0):
        n = n//10
        c = c + 1
    return c

In [3]:
lnnum(26948)

5

In [33]:
import math
def reversee(z):
    rev = 0
    while(z>0):
        rem = z%10
        rev = rem*pow(10,lnnum(z)-1) + rev
        z = z//10
    return rev


In [35]:
reversee(990)

99

In [30]:
def mag(x):
    if(x>reversee(x)):
        d = x-reversee(x)
        if(lnnum(d)==2):
            d = d*10
    else:
        d = reversee(x)-x
        if(lnnum(d)==2):
            d = d*10
    summ = d + reversee(d)
    return summ

In [31]:
mag(351)

1089

In [32]:
m = input("Enter three digit num")
if(len(m)==3 and m[0]!=m[2]):
    print(mag(int(m)))
else:
    print("Invalid Input")

1089
